src/Pos/Launcher/Runner.hs

```diff
@@ -201,6 +202,61 @@ runRealModeDo discoveryCtx transport np@NodeParams {..} sscnp listeners outSpecs
         ntpSlottingVar <- mkNtpSlottingVar
         dlgVar <- RWV.new def
 
+        -- EKG monitoring stuff.
+        --
+        -- Relevant even if monitoring is turned off (no port given). The
+        -- gauge and distribution can be sampled by the server dispatcher
+        -- and used to inform a policy for delaying the next receive event.
+        --
+        -- TODO implement this. Requires time-warp-nt commit
+        --   275c16b38a715264b0b12f32c2f22ab478db29e9
+        -- in addition to the non-master
+        --   fdef06b1ace22e9d91c5a81f7902eb5d4b6eb44f
+        -- for flexible EKG setup.
+        ekgStore <- liftIO $ Metrics.newStore
+        ekgMemPoolSize <- liftIO $ Metrics.createGauge "MemPoolSize" ekgStore
+        ekgMemPoolWaitTime <- liftIO $ Metrics.createGauge "MemPoolWaitTime" ekgStore
+        ekgMemPoolModifyTime <- liftIO $ Metrics.createGauge "MemPoolModifyTime" ekgStore
+        ekgMemPoolQueueLength <- liftIO $ Metrics.createGauge "MemPoolQueueLength" ekgStore
+
+        -- An exponential moving average is used for the time gauges (wait
+        -- and modify durations). The parameter alpha is chosen somewhat
+        -- arbitrarily.
+        -- FIXME take alpha from configuration/CLI, or use a better
+        -- estimator.
+        let alpha :: Double
+            alpha = 0.75
+            -- This TxpMetrics specifies what to do when waiting on the
+            -- mempool lock, when the mempool lock has been granted, and
+            -- when that lock has been released. It updates EKG metrics
+            -- and also logs each data point at debug level.
+            txpMetrics = TxpMetrics
```

Comment:
I would prefer moving to `cardano-sl-txp`, `Runner` shouldn't have non-trivial logic, this logic is specific to transaction processing. Adding `ekg` dependency to `cardano-sl-txp` is fine, because it's already a dependency of `cardano-sl` and I don't think someone will ever want to build `cardano-sl-txp` without `cardano-sl`. These separation into packages is mostly to have some boundaries between logical parts of the code, rather than because small parts (smaller than `cardano-sl` package) are useful by themselves.